packages/@aws-cdk/aws-sagemaker-alpha/lib/model-data.ts

```diff
@@ -6,7 +6,58 @@ import { hashcode } from './private/util';
 
 // The only supported extension for local asset model data
 // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-sagemaker-model-containerdefinition.html#cfn-sagemaker-model-containerdefinition-modeldataurl
-const ARTIFACT_EXTENSION = '.tar.gz';
+const COMPRESSED_ARTIFACT_EXTENSION = '.tar.gz';
+
+/**
+ * Specifies how the ML model data is prepared.
+ */
+export enum CompressionType {
+  /**
+   * If you choose `CompressionType.GZIP` and choose `S3DataType.S3_OBJECT` as the value of `s3DataType`,
+   * S3 URI identifies an object that is a gzip-compressed TAR archive.
+   * SageMaker will attempt to decompress and untar the object during model deployment.
+   */
+  GZIP = 'Gzip',
+  /**
+   * If you choose `CompressionType.NONE` and choose `S3DataType.S3_PREFIX` as the value of `s3DataType`,
+   * S3 URI identifies a key name prefix, under which all objects represents the uncompressed ML model to deploy.
+   *
+   * If you choose `CompressionType.NONE`, then SageMaker will follow rules below when creating model data files
+   * under `/opt/ml/model` directory for use by your inference code:
+   * - If you choose `S3DataType.S3_OBJECT` as the value of `s3DataType`, then SageMaker will split the key of the S3 object referenced by S3 URI by slash (/),
+   *    and use the last part as the filename of the file holding the content of the S3 object.
+   * - If you choose `S3DataType.S3_PREFIX` as the value of `s3DataType`, then for each S3 object under the key name pefix referenced by S3 URI,
+   *    SageMaker will trim its key by the prefix, and use the remainder as the path (relative to `/opt/ml/model`) of the file holding the content of the S3 object.
+   *    SageMaker will split the remainder by slash (/), using intermediate parts as directory names and the last part as filename of the file holding the content of the S3 object.
+   * - Do not use any of the following as file names or directory names:
+   *   - An empty or blank string
+   *   - A string which contains null bytes
+   *   - A string longer than 255 bytes
+   *   - A single dot (.)
+   *   - A double dot (..)
+   * - Ambiguous file names will result in model deployment failure.
+   *    For example, if your uncompressed ML model consists of two S3 objects `s3://mybucket/model/weights` and `s3://mybucket/model/weights/part1`
+   *    and you specify `s3://mybucket/model/` as the value of S3 URI and `S3DataType.S3_PREFIX` as the value of `s3DataType`,
+   *    then it will result in name clash between `/opt/ml/model/weights` (a regular file) and `/opt/ml/model/weights/` (a directory).
```

Comment:
This docstring feels unnecessarily lengthy, and the mention of what file/directory names doesn't seem relevant, at least for the enum. Is there anyway we can add a `@see` tag and link to the documentation instead for all this extra information?